lib/legDetails.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const got = require('got')
+const moment = require('moment-timezone')
+const memoize = require('p-memoize')
+const isString = require('lodash.isstring')
+const isDate = require('lodash.isdate')
+const sortBy = require('lodash.sortby')
+
+const _stations = require('./stations')
+const stations = memoize(_stations, {maxAge: 24*60*60*1000})
+stations()
+
+const findStationByName = (stations, name) => {
+    const station = stations.find(s => s.name === name)
+    if (!station) throw new Error('station not found: '+name)
+    return station
+}
+
+const createStopover = stations => stopover => {
+    const station = findStationByName(stations, stopover.stop)
+    station.location.address = stopover.address
+
+    const arrival = stopover.arrival ? moment.tz(stopover.arrival, 'YYYY-MM-DD HH:mm:ss', station.location.timezone).format() : null
+    const departure = stopover.departure ? moment.tz(stopover.departure, 'YYYY-MM-DD HH:mm:ss', station.location.timezone).format() : null
+
+    const platform = stopover.platform || null
+
+    // todo: stopover.delay
+    const s = ({
+        type: 'stopover',
+        stop: station,
+        arrival,
+        arrivalPlatform: platform,
+        departure,
+        departurePlatform: platform
+    })
+
+    return s
+}
+
+const createLeg = (stations, legId) => stopovers => {
+    const l = ({
+        id: legId,
+        line: {
+            type: 'line',
+            id: stopovers[0].line+'',
+            name: stopovers[0].linetitle,
+            mode: 'bus',
+            public: true,
+            operator: 'ecolines'
+        },
+        origin: findStationByName(stations, stopovers[0].origin),
+        destination: findStationByName(stations, stopovers[0].destination),
+        busPhone: stopovers[0].busphone
+    })
+
+    stopovers = sortBy(stopovers, 'stopsequence')
```

Comment:
`stopovers = stopovers.sort((a, b) => a.stopsequence - b.stopsequence)` is sufficient. Up to you, chose between less 3rd-party code and higher expressiveness.